validator/modules/onnx/__init__.py

```
@@ -0,0 +1,289 @@
+            }
+
+            metrics_dict = self._compute_metrics(
+                ground_truth.flatten(), predictions.flatten(), task_data
+            )
+            return ONNXMetrics(**metrics_dict)
+
+        except Exception as e:
+            if isinstance(e, InvalidTimeSeriesDataException):
+                raise e
+            else:
+                raise ValueError(f"Validation failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)